src/Microsoft.SqlTools.ServiceLayer/AzureFunctions/AzureFunctionsUtils.cs

```diff
@@ -31,19 +34,74 @@ public static IEnumerable<MethodDeclarationSyntax> GetMethodsWithFunctionAttribu
             }
 
             // get all the method declarations with the FunctionName attribute
-            IEnumerable<MethodDeclarationSyntax> methodsWithFunctionAttributes = methodDeclarations.Where(md => md.AttributeLists.Where(a => a.Attributes.Where(attr => attr.Name.ToString().Equals(functionAttributeText)).Any()).Any());
+            IEnumerable<MethodDeclarationSyntax> methodsWithFunctionAttributes = methodDeclarations.Where(md => md.AttributeLists.Where(a => a.Attributes.Where(attr => attr.Name.ToString().Equals(FUNCTION_NAME_ATTRIBUTE_NAME)).Any()).Any());
 
             return methodsWithFunctionAttributes;
         }
 
+        /// <summary>
+        /// Gets the route from an HttpTrigger attribute if specified
+        /// </summary>
+        /// <param name="m">The method</param>
+        /// <returns>The name of the route, or null if no route is specified (or there isn't an HttpTrigger binding)</returns>
+        public static string? GetHttpRoute(this MethodDeclarationSyntax m)
```

Comment:
this is cool! I didn't know about extension methods before 😮

Reply:
Extension methods are the best, I wish other languages had them.